packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/TypeFactory.cs

```diff
@@ -42,8 +43,8 @@ public CSharpType CreateCSharpType(InputType inputType)
             InputUnionType unionType => CSharpType.FromUnion(unionType.VariantTypes.Select(CreateCSharpType).ToArray()),
             InputArrayType listType => new CSharpType(typeof(IList<>), CreateCSharpType(listType.ValueType)),
             InputDictionaryType dictionaryType => new CSharpType(typeof(IDictionary<,>), typeof(string), CreateCSharpType(dictionaryType.ValueType)),
-            InputEnumType enumType => EnumProvider.Create(enumType).Type,
-            InputModelType model => new ModelProvider(model).Type,
+            InputEnumType enumType => new CSharpType(enumType.Name.ToCleanName(), CodeModelPlugin.Instance.Configuration.ModelNamespace, true, true, false, null, null, enumType.Usage.HasFlag(InputModelTypeUsage.Input)),
+            InputModelType model => new CSharpType(model.Name.ToCleanName(), CodeModelPlugin.Instance.Configuration.ModelNamespace, false, false, false, null, null, model.Usage.HasFlag(InputModelTypeUsage.Input)),
```

Comment:
I do not think this makes sense.
What if in the overridden enum or model implementation, the name or namespace has been overridden to be something else?
Here we are assuming the models and enums should take exactly the same name as the input type has, I think we should not make assumptions in such as place, because if something is wrong here, it would be a disaster to debug and find the real issue.

I think here we could just do this:
```suggestion
            InputModelType model => CreateModel(model).Type,
```